synapseclient/models/protocols/access_control_protocol.py

```diff
@@ -145,3 +163,97 @@ def set_permissions(
+        self,
+        recursive: bool = False,
+        include_self: bool = True,
+        include_container_content: bool = False,
+        target_entity_types: Optional[List[str]] = None,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> None:
+        """
+        Delete the Access Control List (ACL) for a given Entity.
+
+        By default, Entities such as FileEntity and Folder inherit their permission from
+        their containing Project. For such Entities the Project is the Entity's 'benefactor'.
+        This permission inheritance can be overridden by creating an ACL for the Entity.
+        When this occurs the Entity becomes its own benefactor and all permission are
+        determined by its own ACL.
+
+        If the ACL of an Entity is deleted, then its benefactor will automatically be set
+        to its parent's benefactor.
+
+        **Special notice for Projects:** The ACL for a Project cannot be deleted, you
+        must individually update or revoke the permissions for each user or group.
+
+        Arguments:
+            include_self: If True (default), delete the ACL of the current entity.
+                If False, skip deleting the ACL of the current entity and only process
+                children if recursive=True or include_container_content=True.
+            recursive: If True and the entity is a container (e.g., Project or Folder),
+                recursively delete ACLs from all child containers and their children.
+                Only works on classes that support the `sync_from_synapse_async` method.
+            include_container_content: If True, delete ACLs from contents directly within
+                containers (files and folders inside Projects/Folders). Defaults to False.
+            target_entity_types: Specify which entity types to process when deleting ACLs.
+                Allowed values are "folder" and "file" (case-insensitive).
+                If None, defaults to ["folder", "file"].
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            None
+
+        Raises:
+            ValueError: If the entity does not have an ID or if an invalid entity type is provided.
+            SynapseHTTPError: If there are permission issues or if the entity already inherits permissions.
+            Exception: For any other errors that may occur during the process.
+
+        Note: The caller must be granted ACCESS_TYPE.CHANGE_PERMISSIONS on the Entity to
+        call this method.
+
+        Example: Delete permissions for a single entity
+            ```python
+            from synapseclient import Synapse
+            from synapseclient.models import File
+
+            syn = Synapse()
+            syn.login()
+
+            File(id="syn123").delete_permissions()
```

Comment:
All entities across synapse like Tables,Views, and etc can have permissions deleted as well.  Is this function only work for files and folders?

Reply:
This function will work for all entity types. What the specific logic in the function is for is dealing with traversing the tree of files and folders within a container (Project/Folder). Users will still be able to delete the permissions for other entities like Tables, Views, ect..

See this script and console log output for an example:
https://gist.github.com/BryanFauble/0b8c0ab8644751241dc7a31e8e779158